tabular/src/autogluon/tabular/configs/config_helper.py

```diff
@@ -0,0 +1,357 @@
+        return self.parent
+
+
+class ConfigBuilder:
+    def __init__(self):
+        self.config = {}
+
+    def presets(self, presets: Union[str, list]) -> ConfigBuilder:
+        """
+        List of preset configurations for various arguments in `fit()`. Can significantly impact predictive accuracy, memory-footprint, and inference latency of trained models, and various other properties of the returned `predictor`.
+        It is recommended to specify presets and avoid specifying most other `fit()` arguments or model hyperparameters prior to becoming familiar with AutoGluon.
+        Available Presets: ['best_quality', 'high_quality_fast_inference_only_refit', 'good_quality_faster_inference_only_refit', 'medium_quality_faster_train', 'optimize_for_deployment', 'ignore_text']
+        It is recommended to only use one `quality` based preset in a given call to `fit()` as they alter many of the same arguments and are not compatible with each-other.
+        If there is an overlap in presets keys, the latter presets will override the earlier ones.
+        """
+        valid_keys = list(tabular_presets_dict.keys())
+        if not isinstance(presets, list):
+            presets = [presets]
+
+        unknown_keys = [k for k in presets if k not in valid_keys]
+        assert len(unknown_keys) == 0, f'The following preset are not recognized: {unknown_keys} - use one of the valid presets: {valid_keys}'
+
+        self.config['presets'] = presets
+        return self
+
+    def time_limit(self, time_limit: int) -> ConfigBuilder:
+        """
+        Approximately how long `fit()` should run for (wallclock time in seconds).
+        If not specified, `fit()` will run until all models have completed training, but will not repeatedly bag models unless `num_bag_sets` is specified.
+        """
+        assert time_limit > 0, 'time_limit must be greater than zero'
+        self.config['time_limit'] = time_limit
+        return self
+
+    def hyperparameters(self, hyperparameters: Union[str, dict]) -> ConfigBuilder:
+        valid_keys = [m for m in MODEL_TYPES.keys() if m not in ['ENS_WEIGHTED', 'SIMPLE_ENS_WEIGHTED']]
+        valid_str_values = list(hyperparameter_config_dict.keys())
+        if isinstance(hyperparameters, str):
+            assert hyperparameters in hyperparameter_config_dict, f'{hyperparameters} is not one of the valid presets {valid_str_values}'
+        elif isinstance(hyperparameters, dict):
+            unknown_keys = [k for k in hyperparameters.keys() if k not in valid_keys]
+            assert len(unknown_keys) == 0, f'The following model types are not recognized: {unknown_keys} - use one of the valid models: {valid_keys}'
+        else:
+            raise ValueError(f'hyperparameters must be either str: {valid_str_values} or dict with keys of {valid_keys}')
+        self.config['hyperparameters'] = hyperparameters
+        return self
+
+    def auto_stack(self, auto_stack=True) -> ConfigBuilder:
```

Comment:
general comment: Should we do type checking in builder or should we entrust that downstream type checks are made?

Reply:
Python type hints are just 'suggestions' even if specified. It really helps when API prevent you to make mistakes by doing strict checks as early as possible.

Reply:
I'm not referring to type hints, but to your explicit assert statements in the builder.